### PR TITLE
update author entity, delete author endpoint, get all authors endpoint and seeder 

### PR DESCRIPTION
**Pull request by irinaenescu2002**

### CODE REVIEW

File: src/main/java/ro/unibuc/hello/service/AuthorService.java [hunk 1]
```diff
@@ -16,19 +21,55 @@ public class AuthorService {
...
+                updateAuthorRequestDto.getDeathDate());
+        var author = authorRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+        author.setDeathDate(updateAuthorRequestDto.getDeathDate());
         return authorRepository.save(author);
     }
 
+    public void deleteAuthor(AuthorDeleteRequestDto authorDeleteRequestDto) {
+        var name = authorDeleteRequestDto.getName();
+        log.debug("Trying to delete author '{}' from system", name);
+        var author = authorRepository.findByName(name);
```
Comment (Qubicon) [inline src/main/java/ro/unibuc/hello/service/AuthorService.java:44]: When deleting an author that has multiple entries with the same name, we encounter the following error:

```Caused by: org.springframework.dao.IncorrectResultSizeDataAccessException: Query { "$java" : Query: { "name" : "Ion Creanga"}, Fields: {}, Sort: {} } returned non unique result.```

We can solve this issue by either searching the author by ID or ensuring that its name is unique.
To make the field unique add:
```@Indexed(unique = true)```

Comment (vgrigoriu) [reply src/main/java/ro/unibuc/hello/service/AuthorService.java:44]: Not that anyone asked me, but people have the annoying habit of not having unique names :-D

Comment (irinaenescu2002) [reply src/main/java/ro/unibuc/hello/service/AuthorService.java:44]: I will solve this bug in a future pull request, at the moment I don't want to block my colleagues

**Comment by Qubicon** [comment]
lgtm (〃￣︶￣)人(￣︶￣〃)

**Review by Qubicon — APPROVED** [review]
lgtm (〃￣︶￣)人(￣︶￣〃)

**Pull request merged**